connect-inject/health_check_resource_test.go

```diff
@@ -0,0 +1,606 @@
+	}
+	return nil
+}
+
+var testPodSpec = corev1.PodSpec{
+	Containers: []corev1.Container{
+		corev1.Container{
+			Name: testPodName,
+		},
+	},
+}
+
+func testServerAgentResourceAndController(t *testing.T, pod *corev1.Pod) (*testutil.TestServer, *api.Client, *HealthCheckResource) {
+	require := require.New(t)
+	// Set up server, client
+	s, err := testutil.NewTestServerConfigT(t, nil)
+	require.NoError(err)
+
+	clientConfig := &api.Config{Address: s.HTTPAddr}
+	require.NoError(err)
+	client, err := api.NewClient(clientConfig)
+	require.NoError(err)
+
+	schema := "http://"
+	consulUrl, err := url.Parse(schema + s.HTTPAddr)
+	require.NoError(err)
+
+	healthResource := HealthCheckResource{
+		Log:                 hclog.Default().Named("healthCheckResource"),
+		KubernetesClientset: fake.NewSimpleClientset(pod),
+		ConsulUrl:           consulUrl,
+		ReconcilePeriod:     0,
+	}
+	return s, client, &healthResource
+}
+
+func registerHealthCheck(t *testing.T, client *api.Client, initialState, reason string) {
+	require := require.New(t)
+	err := client.Agent().CheckRegister(&api.AgentCheckRegistration{
+		Name:      "Kubernetes Health Check",
+		ID:        testHealthCheckID,
+		ServiceID: testServiceNameReg,
+		Notes:     reason,
+		AgentServiceCheck: api.AgentServiceCheck{
+			TTL:    "100000h",
+			Status: initialState,
+			Notes:  reason,
+		},
+	})
+	require.NoError(err)
+}
+
+// We expect to already be pointed at the correct agent
+func getConsulAgentChecks(t *testing.T, client *api.Client) *api.AgentCheck {
+	require := require.New(t)
+	filter := fmt.Sprintf("CheckID == `%s`", testHealthCheckID)
+	checks, err := client.Agent().ChecksWithFilter(filter)
+	require.NoError(err)
+	return checks[testHealthCheckID]
+}
+
+func TestHealthCheckHandlers(t *testing.T) {
+	cases := []struct {
+		Name                 string
+		ValidTests           map[string]bool
+		PreCreateHealthCheck bool
+		InitialState         string
+		Pod                  *corev1.Pod
+		Expected             *api.AgentCheck
+		Err                  string
+	}{
+		{
+			"PodRunning Object Created passing - create check and set passing",
+			getSupportedTestTypes(testTypesBoth),
+			false,
+			api.HealthPassing,
+			&corev1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      testPodName,
+					Namespace: "default",
+					Labels:    map[string]string{labelInject: "true"},
+					Annotations: map[string]string{
+						annotationStatus:  "injected",
+						annotationService: testServiceNameAnnotation,
+					},
+				},
+				Spec: testPodSpec,
+				Status: corev1.PodStatus{
+					HostIP: "127.0.0.1",
+					Phase:  corev1.PodRunning,
+					Conditions: []corev1.PodCondition{{
+						Type:   corev1.PodReady,
+						Status: corev1.ConditionTrue,
+					}},
+				},
+			},
+			&api.AgentCheck{
+				CheckID: testHealthCheckID,
+				Status:  api.HealthPassing,
+				Notes:   testCheckNotesPassing,
+			},
+			"",
+		},
+		{
+			"PodRunning Object Created failed - create check and set failed",
```

Comment:
This test setup seems identical to "PodRunning Object Created passing - create check and set passing".  Is that a mistake?

Reply:
good catch, let me clean this up!
It is either a duplicated test or I just need to flip to Failing ;)